### PR TITLE
fix:[CORE-1262] Violation audit logs API must contain exemption info

### PR DESCRIPTION
**Pull request by ershad-paladin**
# Description

- We want to display the exemption reason and exemption expiry date on violation details screen
- for this, we are adding the exemption reason and exemption expiry date in the /v1/issueauditlog API, so that it can be displayed on UI
- technical detail: on adding an exception, saved the exemption reason and expiry date in the audit logs, so that it can be retrieved along with the audit log that performed it

Fixes # (issue)
Adds exemption detail to audit log in /v1/issueauditlog API

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also
list any relevant details for your test configuration

- [ ] Call the /v2/issue/add-exception API with appropriate request, in postman
- [ ] verify that /v1/issueauditlog API lists the audit logs with exemption reason and expiry date if present
- [ ] test with adding and revoking multiple exemption for the same violation

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] My commit message/PR follows the contribution guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# **Other information**:

List any documentation updates that are needed for the Wiki

### CODE REVIEW

File: api/pacman-api-compliance/src/main/java/com/tmobile/pacman/api/compliance/repository/ComplianceRepositoryImpl.java [hunk 1]
```diff
@@ -2230,7 +2232,8 @@ private StringBuilder formatURL(String assetGroup, String resourcetype, String a
         }
         return urlToQuery;
     }
-    private  String createAuditTrail(String ds, String type, String status, String id,String createdBy, String _type, Map<String, Object> parentDetMap, String target) {
+
+    private String createAuditTrail(String ds, String type, String status, String id, String createdBy, String _type, Map<String, Object> parentDetMap, String target, Map<String, String> optionalAuditFields) {
```
Comment (plyubich) [inline api/pacman-api-compliance/src/main/java/com/tmobile/pacman/api/compliance/repository/ComplianceRepositoryImpl.java:2236]: please extract the params to a CreateAuditTrail model, this param list is getting out of hand and if it keeps growing will be unmanageable.

File: api/pacman-api-compliance/src/main/java/com/tmobile/pacman/api/compliance/repository/ComplianceRepositoryImpl.java [hunk 2]
```diff
@@ -2245,6 +2248,9 @@ private  String createAuditTrail(String ds, String type, String status, String i
         auditTrail.put("_auditdate", date.substring(0, date.indexOf('T')));
         if(parentDetMap != null)
             auditTrail.put(target+RELATIONS, parentDetMap);
+        if (optionalAuditFields != null && !optionalAuditFields.isEmpty()) {
```
Comment (plyubich) [inline api/pacman-api-compliance/src/main/java/com/tmobile/pacman/api/compliance/repository/ComplianceRepositoryImpl.java:2251]: I believe isEmpty() will also check for null?

File: api/pacman-api-compliance/src/main/java/com/tmobile/pacman/api/compliance/repository/ComplianceRepositoryImpl.java [hunk 3]
```diff
@@ -3017,9 +3026,13 @@ public ExemptionResponse createOrRevokeUserExemptionRequest(ExemptionRequest exe
...
+                    optionalAuditFields.put(Constants.ISSUE_EXEMPTION_REASON, exemptionRequest.getExceptionReason());
+
                     builderRequestAudit.append(String.format(ACTION_TEMPLATE_AUDIT, dataSource, id))
                             .append(createAuditTrail(exemptionRequest.getAssetGroup(), targetType,exemptionRequest.getAction().equals(ExemptionActions.CREATE_EXEMPTION_REQUEST)?REQUEST_EXEMPT:exemptionRequest.getAction().equals(ExemptionActions.REVOKE_EXEMPTION_REQUEST)?REVOKE_EXEMPT:exemptionRequest.getAction().equals(ExemptionActions.CANCEL_EXEMPTION_REQUEST)?DENY_EXEMPT:GRANT_EXEMPT, id,
-                                    exemptionRequest.getAction().equals(ExemptionActions.APPROVE_EXEMPTION_REQUEST)?exemptionRequest.getApprovedBy():exemptionRequest.getCreatedBy(), _type, parentDetMap, target)+"\n");
+                                    exemptionRequest.getAction().equals(ExemptionActions.APPROVE_EXEMPTION_REQUEST)?exemptionRequest.getApprovedBy():exemptionRequest.getCreatedBy(), _type, parentDetMap, target, optionalAuditFields)+"\n");
```
Comment (plyubich) [inline api/pacman-api-compliance/src/main/java/com/tmobile/pacman/api/compliance/repository/ComplianceRepositoryImpl.java:3035]: needs to format code.

**Review by plyubich — CHANGES_REQUESTED** [review]
Minor changes needed.

File: api/pacman-api-compliance/src/main/java/com/tmobile/pacman/api/compliance/repository/ComplianceRepositoryImpl.java (hunk 1 above)
Comment (ershad-paladin) [reply api/pacman-api-compliance/src/main/java/com/tmobile/pacman/api/compliance/repository/ComplianceRepositoryImpl.java]: done

File: api/pacman-api-compliance/src/main/java/com/tmobile/pacman/api/compliance/repository/ComplianceRepositoryImpl.java (hunk 3 above)
Comment (ershad-paladin) [reply api/pacman-api-compliance/src/main/java/com/tmobile/pacman/api/compliance/repository/ComplianceRepositoryImpl.java]: done

**Review by plyubich — APPROVED** [review]

**Review by santhosh-challa — APPROVED** [review]

**Pull request merged**